src/include/migraphx/generic_float.hpp

```diff
@@ -373,7 +531,7 @@ template <unsigned int M, unsigned int E, unsigned int F>
 class numeric_limits<migraphx::generic_float<M, E, F>>
 {
     public:
-    static constexpr bool has_infinity = true;
+    static constexpr bool has_infinity = not(M == 0 and E == 0);
```

Comment:
Tidy is being nitpicky here but It saves you the extra not if you make this  `(M == 0 or E == 0)`